core/base/src/TObject.cxx

```diff
@@ -778,6 +779,36 @@ void TObject::SavePrimitiveConstructor(std::ostream &out, TClass *cl, const char
    out << variable_name << " = new " << cl->GetName() << "(" << constructor_agrs << ");" << std::endl;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Save array in the output stream "out".
+/// Create unique variable name based on prefix value
+
+TString TObject::SavePrimitiveArray(std::ostream &out, const char *prefix, Int_t len, Double_t *arr)
+{
+   thread_local int arrid = 0;
+
+   TString arrname = TString::Format("%s_darr%d", prefix, arrid++);
+
+   out << "   Double_t " << arrname << "[" << len << "] = { ";
+   if (len > 0) {
+      const auto old_precision{out.precision()};
+      constexpr auto max_precision{std::numeric_limits<double>::digits10 + 1};
+      out << std::setprecision(max_precision);
+      for (Int_t i = 0; i < len-1; i++) {
+         out << arr[i] << ",";
+         if (i && (i % 16 == 0))
+            out << std::endl << "   ";
```

Comment:
This should use `"\n"` instead of `std::endl` (we don't need to flush the stream every time)

Reply:
I do not think that performance here is a major concern. 
For me `std::endl` is preferable while it is standard line ending marker.

Reply:
> For me std::endl is preferable while it is standard line ending marker.

Please read the [std::endl documentation](https://en.cppreference.com/w/cpp/io/manip/endl) before dismissing my point: `std::endl` has nothing to do with being a "standard line ending marker" (whatever that means) but it specifically puts a `\n` followed by a `ostream.flush()`. As such, the documentation itself suggests not using it unless you specifically want to flush the output stream (which only makes sense at the very end of the output - if at all)

Reply:
Performance here not a problem at all.
Even writing 1^10 array entries here will be done in reasonable time.
Real problem is in many other ROOT classes - like TPolyMarker from this PR.
Without usage of this method macro execution for just 50K lines takes 37s or even 110s if compiled with gcc.
And in ROOT exactly 13987 places where `std::endl` is used. 
To be consequent - one need to review all these places and replace approx 13800 or maybe 13900 of them by "\n"

Reply:
This particular instance (and most of the 13987 others) are not particularly important in themselves.  However when writing new code or updating existing code, I do recommend to use `\n` to introduce new lines and `std::endl` only when the intent is to flush the output.